File: pset6/mario/more/mario.py
import cs50

# asks user for the height of the pyramid.
pyramidheight = cs50.get_int("Height: ")
# asks user the height of the pyramid again if height is not 1 and 8 inclusive.
while pyramidheight < 1 or pyramidheight > 8:
    pyramidheight = cs50.get_int("Height: ")

# prints pyramid based on height.
for i in range(0, pyramidheight):
    # prints spaces while printing a row of pyramid.
    for ls in range(pyramidheight-1, i, -1):
        print(" ", end="")
    # prints left hashes while printing a row of pyramid.
    for lh in range(0, i + 1):
        print("#", end="")
    # prints two spaces at the middle while printing a row of pyramid.
    print("  ", end="")
    # ptints right hashes while printing a row of pyramid.
    for rh in range(0, i + 1):
        print("#", end="")
    # prints next line at the end of a row of pyramid.
    print()

File: pset1/mario/less/mario.c
#include<cs50.h>
#include<stdio.h>

void printpyramid(int pyramidheight);

int main(void)
{
    int height;

    //ask user for height between 1 to 8 to print pyramid.
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);
    //printf("Stored: %i\n",height);


    printpyramid(height);
}

//prints pyramid based on height.
void printpyramid(int pyramidheight)
{
    //prints a row of the pyramid.
    for (int i = 0; i < pyramidheight; i = i + 1)
    {
        /*for(int j=0;j<i;j=j+1){
            printf("#");
        }*/
        /*for(int d=pyramidheight-1;d>i;d=d-1){
            printf(".");
        }*/

        //adds space(s) to a row of the pyramid while printing.
        for (int s = pyramidheight - 1; s > i; s = s - 1)
        {
            printf(" ");
        }

        //adds hashes(s) to a row of the pyramid while printing.
        for (int h = 0; h <= i; h = h + 1)
        {
            printf("#");
        }

        //adds next line to a row of the pyramid while printing.
        printf("\n");
    }
}